platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/SwirldState.java

```diff
@@ -102,6 +103,19 @@ default void preHandle(final Event event) {}
      */
     void handleConsensusRound(final Round round, final SwirldDualState swirldDualState);
 
+    /**
+     * Update the stake of the nodes in the address book
+     *
+     * @param configAddressBook the address book as loaded from config.txt. This address book may contain new nodes not
+     *                          present in the stateAddressBook. Must not be null.
+     * @param stateAddressBook  the address book loaded from the latest state on disk, or null if starting from
+     *                          genesis.  May be null.
+     * @return a copy of the configuration address book with updated stake.
+     */
+    default AddressBook updateStake(final AddressBook configAddressBook, final AddressBook stateAddressBook) {
```

Comment:
if this is called on the state, why are we passing in the state address book?

Reply:
Great Question.   I asked @cody-littley, and he indicated that the state's address book used to be in the application's state, but now it is in the platform's state and the application no longer has access unless we give it to it.

Reply:
To me it seems like if services should update the AB with stake, it should only be given one AB to update

Reply:
I have removed the second argument.